math/src/main/java/com/powsybl/math/timeseries/TimeSeriesTable.java

```diff
@@ -0,0 +1,597 @@
+        }
+    }
+
+    private int fromVersion;
+
+    private int toVersion;
+
+    private List<TimeSeriesMetadata> timeSeriesMetadata;
+
+    private TimeSeriesIndex tableIndex;
+
+    private final TIntArrayList timeSeriesIndexDoubleOrString = new TIntArrayList(); // global index to typed index
+
+    private final TimeSeriesNameMap doubleTimeSeriesNames = new TimeSeriesNameMap();
+
+    private final TimeSeriesNameMap stringTimeSeriesNames = new TimeSeriesNameMap();
+
+    private DoubleBuffer doubleBuffer;
+
+    private CompactStringBuffer stringBuffer;
+
+    // statistics
+    private double[] means;
+
+    private double[] stdDevs;
+
+    public TimeSeriesTable(int fromVersion, int toVersion) {
+        TimeSeriesIndex.checkVersion(fromVersion);
+        TimeSeriesIndex.checkVersion(toVersion);
+        if (toVersion < fromVersion) {
+            throw new TimeSeriesException("To version is expected to be greater tha from version");
+        }
+        this.fromVersion = fromVersion;
+        this.toVersion = toVersion;
+    }
+
+    private void checkIndex(TimeSeries timeSeries) {
+        if (timeSeries.getMetadata().getIndex() != InfiniteTimeSeriesIndex.INSTANCE) {
+            if (tableIndex == null) {
+                tableIndex = timeSeries.getMetadata().getIndex();
+            } else {
+                if (!tableIndex.equals(timeSeries.getMetadata().getIndex())) {
+                    throw new TimeSeriesException("All time series must be synchronized");
+                }
+            }
+        }
+    }
+
+    private void initTable(List<DoubleTimeSeries> doubleTimeSeries, List<StringTimeSeries> stringTimeSeries) {
+        if (timeSeriesMetadata != null) {
+            return; // already initialized
+        }
+
+        timeSeriesMetadata = new ArrayList<>(doubleTimeSeries.size() + stringTimeSeries.size());
+
+        for (DoubleTimeSeries timeSeries : doubleTimeSeries) {
+            checkIndex(timeSeries);
+            timeSeriesMetadata.add(timeSeries.getMetadata());
+            int i = doubleTimeSeriesNames.add(timeSeries.getMetadata().getName());
+            timeSeriesIndexDoubleOrString.add(i);
+        }
+        for (StringTimeSeries timeSeries : stringTimeSeries) {
+            checkIndex(timeSeries);
+            timeSeriesMetadata.add(timeSeries.getMetadata());
+            int i = stringTimeSeriesNames.add(timeSeries.getMetadata().getName());
+            timeSeriesIndexDoubleOrString.add(i);
+        }
+
+        if (tableIndex == null) {
+            throw new TimeSeriesException("None of the time series have a finite index");
+        }
+
+        int versionCount = toVersion - fromVersion + 1;
+
+        // allocate double buffer
+        int doubleBufferSize = versionCount * doubleTimeSeriesNames.size() * tableIndex.getPointCount();
+        doubleBuffer = createDoubleBuffer(doubleBufferSize);
+        for (int i = 0; i < doubleBufferSize; i++) {
+            doubleBuffer.put(Double.NaN);
+        }
```

Comment:
Create another version of createDoubleBuffer which also fill the buffer:
doubleBuffer = createDoubleBuffer(doubleBufferSize, Double.Nan);

Reply:
Done